optica.rb

```diff
@@ -13,8 +13,17 @@ class Optica < Sinatra::Base
 
   # endpoint for fab usage
   get '/roles' do
-    # keys must be strings
-    return get_nodes(request, ['role', 'id', 'hostname'])
+    params_to_fetch = ['role', 'id', 'hostname']
+    params = CGI::parse(request.query_string)
+    if params['_additional_fields']
```

Comment:
Maybe just `_fields` for brevity? Although "additional" is clearer since we're adding to the default set.